modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/components/PageRenderer/index.js

```diff
@@ -184,6 +186,16 @@ const Renderer = ({
 		PAGE_HEADER_COMPONENT_TYPE[page.headerRenderer] ||
 		Components.PageHeader;
 
+	const hasRequiredIconDescription =
+		!portletNamespace.includes('Admin') &&
+		page.rows.filter(
+			(x) =>
+				x &&
+				x.columns[0] &&
+				x.columns[0].fields[0] &&
+				x.columns[0].fields[0].required
+		).length > 0;
```

Comment:
@alinedoleron, do you think we can use the `pageVisitor` here or it's only available on MetalJS for now?

Reply:
Hi @natocesarrego. Yes, we can. I've seen pagesVisitor being used recently after migrating ddm-form-renderer to React.

Reply:
Hi all,

So do you refer to use fieldReducer.es.js to check if we have a field required? 

Regards.

Reply:
You could use something like:

```
let hasFieldRequired = false;
const visitor = new PagesVisitor(pages);
   visitor.mapFields(function (field) {
        if (field.required) {
          hasFieldRequired = true;
        }
      });
```

Reply:
Hi,
Thanks for the example but I get this dependency error:

Reply:
Hi George, sorry for not telling the correct path. Actually you should do:
`import {PagesVisitor} from '../../util/visitors.es';` to reach the file.

Reply:
Thanks! it worked, I pushed my changes, please review.
Regards.

Reply:
Thank you very much for your amazing input, @alinedoleron! 😄 👏